### PR TITLE
DSA: Use seat nonbid code 300

### PR DESCRIPTION
**Pull request by bsardo**
Seat nonbid code 305 has not been formally approved yet so we will use 300 until then.
This PR also adds a DSA specific warning code as I don't think the invalid bid response error code makes sense since this is not a server error nor is it a malformed bid response. Instead it is a validation error.

### CODE REVIEW

**Review by VeronikaSolovei9 — APPROVED** [review]

File: exchange/exchange.go [hunk 1]
```diff
@@ -1247,12 +1247,12 @@ func (e *exchange) makeBid(bids []*entities.PbsOrtbBid, auc *auction, returnCrea
 	for _, bid := range bids {
 		if !dsa.Validate(bidRequest, bid) {
 			RequiredError := openrtb_ext.ExtBidderMessage{
-				Code:    errortypes.BadServerResponseErrorCode,
-				Message: "bidResponse rejected: DSA object missing when required",
+				Code:    errortypes.InvalidBidResponseDSAWarningCode,
+				Message: "bid response rejected: object missing when required",
```
Comment (SyntaxNode) [inline exchange/exchange.go:1251]: Why did you remove DSA from the message?

File: exchange/non_bid_reason.go [hunk 2]
```diff
@@ -6,9 +6,9 @@ package exchange
 type NonBidReason int
 
 const (
-	NoBidUnknownError                      NonBidReason = 0   // No Bid - General
+	NoBidUnknownError                      NonBidReason = 0 // No Bid - General
+	ResponseRejectedGeneral                NonBidReason = 300
 	ResponseRejectedCategoryMappingInvalid NonBidReason = 303 // Response Rejected - Category Mapping Invalid
-	ResponseRejectedPrivacy                NonBidReason = 305
```
Comment (SyntaxNode) [inline exchange/non_bid_reason.go:11]: Thanks. I'm working on making 305 a thing, but right now it's not defined in the spec. I expect we can change this to 305 the following release.

File: exchange/exchange.go (hunk 1 above)
Comment (bsardo) [reply exchange/exchange.go:1251]: I figured the code gives the context. Do you want me to add it back?

Comment (bsardo) [reply exchange/exchange.go]: Discussed offline. Added it back for human readability.

**Review by SyntaxNode — APPROVED** [review]
